### ISSUE TITLE
Error Message on condition basis and show message on field if validation passed

### ISSUE THREAD

**Issue by shivam091** [horprogs/Just-validate#35]
Hi,

I came across a situation that I need to return error message on conditions basis. At present, I tried with one option `validator` present in library but I could not use it because it returns only one error message when function returns falsy value.

Just for an example to let you understand about the problem:

```javascript

validator.addField("[name='model[field_name]']", [
  {
    rule: "required",
    errorMessage: "Field is required"
  },
  {
    validator: (fieldValue, fields) => {
      if (value < 5) {
        return false;  // different message than one specified in below errorMessage.
      } else if (value > 10) {
        return false;  // different message than one specified in below errorMessage.
      } else {
        return true; // Success message in green popover or tooltip
      }
    },
    errorMessage: "Invalid"
  }
]);

```

At present, I think, it is not supported to return / display message if field passed validations and somewhere in huge applications, there might be such a scenario where we need to show success message somehat like "Looks great!", etc.

and also I think, there is no provision to show different error messages on condition basis.

Dear owner, what do you think on above two scenarios ?

**Comment by horprogs** [comment]
Thanks for the suggestions, actually I was thinking about conditional messages, I'll add it, also I'll think about adding something like `successMessage`

**Comment by protaskz-in** [comment]
Welcome ! Something customizable like we have in case of errors that we can show them in our own styles, may be inline text or may be on tooltip. then some additional class to field like "is-valid" where we can set success icon to the field, just like we have form fields having errors. Something like just a replica of options we have for invalid fields.

Well, thank you for updating the plugin with issues' fixes and Wish you a very happy new year :)

**Comment by horprogs** [comment]
Happy New Year! Published `3.3.0` version. It includes conditional messages and success messages. Success message could be customized by defining `successFieldStyle`, `successFieldCssClass`, `successLabelStyle`, `successLabelCssClass` 
as we do for error messages.  Usage example:

```
  validator.addField(
    "[name='model[field_name]']",
    [
      {
        rule: 'required',
        errorMessage: 'Field is required',
      },
      {
        validator: (fieldValue, fields) => {
          if (fieldValue < 5) {
            return false;
          } else if (fieldValue > 10) {
            return false;
          } else {
            return true;
          }
        },
        errorMessage: (fieldValue, fields) => {
          if (fieldValue < 5) {
            return 'One error message';
          } else if (fieldValue > 10) {
            return 'Another error message';
          } else {
            return 'Default error message';
          }
        },
      },
    ],
    {
      successMessage: 'Looks good',
    }
  );
```

https://just-validate.dev/examples/#success-label
